setuptools/dist.py

```diff
@@ -366,8 +366,10 @@ def __init__(self, attrs=None):
         # prime it here from our value if not automatically set
         self.metadata.project_urls = getattr(
             self.metadata, 'project_urls', self.project_urls)
-        self.metadata.long_description_content_type = attrs.get(
-            'long_description_content_type'
+        # Some initalizers may pre-set this, otherwise use from attr
+        self.metadata.long_description_content_type = getattr(
+            self.metadata, 'long_description_content_type',
+            attrs.get('long_description_content_type')
```

Comment:
Does it make sense to guard against the case where `self.metadata.long_description_content_type` might be  `None` but `attrs.get('long_description_content_type')` has some value? I'm not sure how that could happen but if it did, the actual value would be ignored with this change.